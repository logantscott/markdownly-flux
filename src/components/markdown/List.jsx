import React from 'react';
import { useHistory } from 'react-router-dom';
import { useSelector, useDispatch } from '../../hooks/markdownContext';
import { getFiles } from '../../selectors/markdownSelectors';
import { updateCurrentFileID } from '../../actions/markdownActions';


const List = () => {
  const dispatch = useDispatch();
  const history = useHistory();

  const handleEdit = ({ target }) => {
    dispatch(updateCurrentFileID(target.value));
    history.push('/editor');
  };

  const files = Object.values(useSelector(getFiles));

  const fileElements = files.map(({ title, body, id }) => (
    <li key={id}>
      <button onClick={handleEdit} value={id}>Edit {title}</button>
      <h2>{title}</h2>
      <pre>{body.substring(0, 150)}</pre>
    </li>
  ));
  return (
    <ul>
      {fileElements}   
    </ul>
  );
};

export default List;
