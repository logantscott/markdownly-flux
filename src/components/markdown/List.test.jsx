import React from 'react';
import { shallow } from 'enzyme';
import List from './List';
import AppProvider from '../../providers/appProvider';

describe('test the list component', () => {
  it('renders the list', () => {
    const wrapper = shallow(
      <AppProvider>
        <List />
      </AppProvider>
    );
    expect(wrapper).toMatchSnapshot();
  });
  
});



