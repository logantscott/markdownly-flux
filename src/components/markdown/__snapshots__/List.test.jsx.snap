// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test the list component renders the list 1`] = `
<ContextProvider
  value={
    Object {
      "dispatch": [Function],
      "state": Object {
        "files": Array [
          Object {
            "body": "this is a markdown file",
            "id": "ID2938248",
            "title": "test",
          },
          Object {
            "body": "another markdown",
            "id": "ID392045",
            "title": "another-test",
          },
          Object {
            "body": "last test file",
            "id": "ID24558248",
            "title": "one-more",
          },
        ],
        "markdown": "# Hi there",
      },
    }
  }
>
  <List />
</ContextProvider>
`;
